docs/lang/articles/contribution/dev_install.md

```diff
@@ -87,224 +108,29 @@ installer.
-  python3 -m pip install --user numpy GitPython coverage colorama autograd
-  ```
-
-2. Make sure you have LLVM 10.0.0. Note that Taichi uses a **customized
-  LLVM** so the pre-built binaries from the LLVM official website or
-  other sources probably won't work. Here we provide LLVM binaries
-  customized for Taichi, which may or may not work depending on your
-  system environment:
-  - [LLVM 10.0.0 for macOS](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/taichi-llvm-10.0.0-macos.zip)
-
-- If the downloaded LLVM does not work, please build from source:
-    ```bash
-    wget https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/llvm-10.0.0.src.tar.xz
-    tar xvJf llvm-10.0.0.src.tar.xz
-    cd llvm-10.0.0.src
-    mkdir build
-    cd build
-    cmake .. -DLLVM_ENABLE_RTTI:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=OFF -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_ENABLE_TERMINFO=OFF
-    # If you are building on NVIDIA Jetson TX2, use -DLLVM_TARGETS_TO_BUILD="ARM;NVPTX"
-
-    make -j 8
-    sudo make install
 
-    # Check your LLVM installation
-    llvm-config --version  # You should get 10.0.0
-    ```
 ### Setting up Taichi for development
 
-1. Set up environment variables for Taichi:
-
-  - Please add the following script to your rc file
-    (`~/.bashrc`, `~/.zshrc` or etc. , same for other occurrences in
-    this guide):
-
-    ```bash
-    export TAICHI_REPO_DIR=/path/to/taichi  # Path to your taichi repository
-    export PYTHONPATH=$TAICHI_REPO_DIR/python:$PYTHONPATH
-    export PATH=$TAICHI_REPO_DIR/bin:$PATH
-    # export CXX=/path/to/clang  # Uncomment if you encounter issue about compiler in the next step.
-    # export PATH=/opt/llvm/bin:$PATH  # Uncomment if your llvm or clang is installed in /opt
-    ```
-
-  - To reload shell config, please execute
-
-    ```bash
-    source ~/.bashrc
-    ```
-
-    :::note
-    If you're using fish, use `set -x NAME VALUES`, otherwise it
-    won't be loaded by child processes.
-    :::
-
-2. Clone the taichi repo **recursively**, and build:
+1. Clone the Taichi repo **recursively**, and build:
 
   ```bash
-  git clone https://github.com/taichi-dev/taichi --depth=1 --branch=master
+  git clone --recursive https://github.com/taichi-dev/taichi --depth=1 --branch=master
   cd taichi
-  git submodule update --init --recursive --depth=1
-  mkdir build
-  cd build
-  cmake ..
-  # On macOS, if you do not set clang as the default compiler
-  # use the line below:
-  #   cmake .. -DCMAKE_CXX_COMPILER=clang
-  #
-  # Alternatively, if you would like to set clang as the default compiler
-  # On Unix CMake honors environment variables $CC and $CXX upon deciding which C and C++ compilers to use
-  make -j 8
+  # export CXX=/path/to/clang  # Uncomment if clang is not system default compiler.
+  python3 setup.py develop  # Optionally add DEBUG=1 to keep debug information.
```

Comment:
Do you think we should give the solusion when permission denied problems shown? Because I know dunfan and mingrui have the same questions this morning.

Solusion:

1. python setup.py develop --user
2. install conda and use conda env

Reply:
Yea nice catch! I added `--user` to our default install instruction, and also added an entry in troubleshooting just in case. :D